FpsCheck.py

```diff
@@ -409,51 +409,47 @@ def find_invalid_alias_eSLDs(self, check_sets):
         members of icann
         Reads the ccTLDs and makes sure that they match their equivalent sites,
-        and that they're domains are part of ICANNS list of country codes.
-        If either of these is not the case, appends an error to the error_list
+        and that their eTLDs are part of ICANN's list of country codes.
+        If either of these is not the case, appends an error to the error_list.
+        Note: A site may list a variant with "com" as its eTLD IFF the site 
+        being aliased has an eTLD on ICANN's list of countrycodes. 
         Args:
-            check_sets: a dictionary of primary->FpsSet
+            check_sets: Dict[string, FpsSet]
         Returns:
             None
         """
-        for primary in check_sets:
-            curr_set = check_sets[primary]
-            if curr_set.ccTLDs:
-                for aliased_site in curr_set.ccTLDs:
-                    # first check if the aliased site is actually anywhere else
-                    # in the fps
-                    if aliased_site != primary:
-                        if curr_set.associated_sites:
-                            sites = curr_set.associated_sites 
-                        else:
-                            sites = []
-                        if curr_set.service_sites:
-                            sites += curr_set.service_sites
-                        if aliased_site not in sites:
-                            self.error_list.append(
-                                "The aliased site " + aliased_site + 
-                                " contained within the ccTLDs must be a " +
-                                "primary, associated site, or service site " +
-                                "within the firsty pary set for " + primary)
-                    # check the validity of the aliases
-                    aliased_domain, aliased_tld = aliased_site.split(".", 1)
-                    if aliased_tld in self.icanns:
-                        icann_check = self.icanns.union({"com"})
-                    else:
-                        icann_check = self.icanns
-                    plus1s = [(site, site.split(".")[0], site.split(".")[-1])
-                              for site in curr_set.ccTLDs[aliased_site]]
-                    for eSLD in plus1s:
-                        if eSLD[1] != aliased_domain:
-                            self.error_list.append(
-                                "The following top level domain must match: " 
-                                + aliased_site + ", but is instead: " 
-                                + eSLD[0])
-                        if eSLD[2] not in icann_check:
-                            self.error_list.append(
-                                "The provided country code: " + eSLD[2] + 
-                                ", in: " + eSLD[0] + 
-                                " is not a ICANN registered country code")
+        for primary, curr_set in check_sets.items():
```

Comment:
Food for thought: the `primary` is also accessible via `curr_set.primary`. So, it seems that it would be fine to accept a `List[FpsSet]`, rather than a `Dict[string, FpsSet]`, since we're not relying on the fact that it's a dictionary.

In general, we should code to the interface we need; we shouldn't impose structure that we don't actually care about (if we can avoid it, anyway).

Reply:
SGTM, I didn't mean to imply it should be done in this PR!